stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -507,13 +507,26 @@ async def all_catalogs(
                 break
             token = next_token
 
+        if catalog_path:
+            catalog_url = f"catalogs/{catalog_path}"
+        else:
+            catalog_url = ""
+
         links = [
-            {"rel": Relations.root.value, "type": MimeTypes.json, "href": base_url},
-            {"rel": Relations.parent.value, "type": MimeTypes.json, "href": base_url},
+            {
+                "rel": Relations.root.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
+            },
+            {
+                "rel": Relations.parent.value,
+                "type": MimeTypes.json,
+                "href": urljoin(base_url, catalog_url),
+            },
             {
                 "rel": Relations.self.value,
                 "type": MimeTypes.json,
-                "href": urljoin(base_url, "catalogs"),
+                "href": f"{urljoin(base_url, catalog_url)}/catalogs",
```

Comment:
Shouldn't this be just `urljoin(base_url, catalog_url)`?

Reply:
This is for the `/catalogs` endpoint itself, not for each individual catalog.

Reply:
You're correct here as well.